docs/instructlab-cli-1.0.0.md

```diff
@@ -0,0 +1,70 @@
+running the LAB methodology created by IBM Research. As the project evolved and the organization looked into creating a proper PyPI package for it, the decision was made
+to rename the repo to `instructlab/instructlab` to keep the repo name consistent with the PyPI package name. The rest of this document will being using "InstructLab" to
+refer to this repo and Python package.
+
+Today, InstructLab has gone from a scrappy research project to an upstream community serving as the basis for multiple downstreams, with the goal
+to continuing to evolve the community to encourage more participation from additional stakeholders. To wit, it would behoove us to determine what exactly we should be
+roadmapping between now and a proper 1.0.0 release, which demonstrates the following to existing and potential community members:
+
+1. An official goalpost for the community denoting the evolution of InstructLab from a pre-1.0 project lacking the stability and supportability typically seen from 1.0-and-beyond projects.
+1. A dedicated set of V1 interfaces that can be counted on for continuous usage of InstructLab 1.0 with future provisions made for backwards compatibility for subsequent Y-Streams and Z-Streams.
+1. A commitment from the Oversight Committee and Maintainer teams to continue to maintain InstructLab throughout a 1.y cycle and work towards an eventual 2.0.
+
+## MVP for an InstructLab 1.0.0
+
+At a high-level, these are the items the Maintainer teams believe should serve as prereqs for releasing an InstructLab 1.0.0:
+
+### A consistent naming scheme
+
+As noted in the `Context and Goals` section, InstructLab started off as just as a CLI - however, we are planning for this package to serve as a more general "Engine" -
+being a place where a future REST API can be defined that is used by both the CLI aspect as well as an official GUI for orchestrating the entire LAB workflow. Despite
+this, the repo is often still referred to as "the CLI". We as an organization need a better term to refer to this repo as, and should adopt the relevant documentation
+and meetings accordingly.
+
+### A fully-realized configuration scheme, centered around the usage of system profiles
+
+The InstructLab configuration scheme has transformed in many ways since the project's inception, from the `config.yaml` file that initially served as the user's config,
+to the addition of code-based Pydantic defaults, to train profiles, to system profiles. We need to fully-decouple this config from the Click library, remove the need for
+a `config.yaml` file, and have a consistent config scheme that can be easily extended.
+
+### An official v1 REST API schema
+
+We need to have a defined v1 REST API schema - while this does not preclude future updates, something mature enough to serve as a v1 API throughout subsequent Y-Streams
+for an InstructLab 1.0 is a must for such a milestone.
+
+### Integration of InstructLab with RAG
```

Comment:
We should probably better define what this actually means.

Reply:
+1

Reply:
I would love for the folks working on RAG to chime in here and suggest more detail, I've been pretty far removed from that work

Reply:
I am going to resolve this convo for now since no SMEs have chimed in - we can flesh this out in a followup